src/Simplify.elm

```diff
@@ -813,6 +813,9 @@ Destructuring using case expressions
     Set.union set Set.empty
     --> set
 
+    Set.union (Set.fromList [ a, b ]) (Set.fromList [ c, d ])
+    --> Set.fromList [ a, b, c, d]
```

Comment:
Do `Set.union` and `Dict.union` really behave differently?
Looking at their implementation, it doesn't seem like it: https://github.com/elm/core/blob/1.0.5/src/Set.elm#L100

Reply:
No, it's more that `Dict.union` needs to have things inserted in the opposite order than `Array.append`/`List.append`, otherwise you ge tbehavior changes if you have
```elm
Dict.union (Dict.fromList [ ( "a", 1 ) ]) (Dict.fromList [ ( "a", 2 ) ])
-- results in Dict.fromList [ ( "a", 1 ) ]
-- so we should simplify it to
--> Dict.fromList [ ( "a", 2 ), ( "a", 1 ) ]

-- If we simplified it to
Dict.fromList [ ( "a", 1 ), ( "a", 2 ) ]
-- then the value of that would be 
-- Dict.fromList [ ( "a", 2 ) ]
```
So we need to reverse it.

`Set` behaves just like `Dict`, except that it is is a special case of `Dict` where the order really doesn't matter. So I chose to keep the same behavior as `Array.append`/`List.append` which I think feels a lot more natural.